friends/src/App.js

```diff
@@ -0,0 +1,140 @@
+import React, { Component } from 'react';
+import { Route, withRouter } from 'react-router-dom';
+import axios from 'axios';
+
+import FriendList from './components/FriendList';
+import NewFriend from './components/NewFriend';
+import EditFriend from './components/EditFriend';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+        friends: [],
+        currentFriend: []
+    };
+
+    this.addNewFriendHandleSubmit = this.addNewFriendHandleSubmit.bind(this);
```

Comment:
These can all go away by converting your class methods to `arrow functions` | `greenArrow = () => {}`